src/contract.ts

```diff
@@ -244,7 +380,7 @@ export class Contract {
    * @param inputs  - list of inputs(fields) that are in the abi
    * @return {Calldata} - parsed arguments in format that contract is expecting
    */
-  private compileCalldata(args: Args, inputs: AbiEntry[]): Calldata {
+  compileCalldata(args: Args, inputs: AbiEntry[]): Calldata {
```

Comment:
Should compileCalldata receive the full ABI to be able to compile Structs aswell? or is that handled somewhere else?